drud/ddev
# Add segment to instrumentation, fixes #1640, fixes #1344

## The Problem/Issue/Bug:

#1640 requests adding instrumentation pointed at Segment.

This adds it, with a starting set of attributes/properties and a hashed hostID as the user id.

## How this PR Solves The Problem:

Here's what's currently being delivered to Segment:
```
client.Track(&analytics.Track{
  UserId: "ccb22589fa984b1772a5bccd8972bb19e267337f175178566339258146433942",
  Event: "start",
  Properties: map[string]interface{}{
    "bgsyncimg": "drud/ddev-bgsync:v1.9.0",
    "dbaimg": "drud/phpmyadmin:v1.9.0",
    "dbimg": "drud/ddev-webserver:20190703_klonos_backdrop",
    "dockerComposeVersion": "1.24.0",
    "dockerVersion": "19.03.0-rc2",
    "isDockerToolbox": false,
    "php_version": "7.2",
    "router_http_port": "80",
    "router_https_port": "443",
    "router_status": "healthy",
    "ssh_agent_status": "healthy",
    "status": "running",
    "type": "drupal8",
    "version": "v1.9.1-8-g4cabde3c",
    "webimg": "drud/ddev-webserver:20190703_klonos_backdrop",
    "webserver_type": "nginx-fpm",
  },
})
```

## Manual Testing Instructions:

Download one or more of the build artifacts and experiment with various ddev commands.
* You should see the user and the event represented in Segment (debug tab)
* The event should have what you want in it.
* Sentry should also show the same event (and it has ServerHash in it to correlate the two services)

## Automated Testing Overview:

No testing was added here. 

## Related Issue Link(s):

#1344 requested removing some sensitive data from Sentry, this removes at least some. I couldn't figure out how to get rid of server_name with the current API, maybe in the future with https://github.com/drud/ddev/issues/1703
#1640 "Replace sentry with segment"

## Release/Deployment notes:

* We need to set the Sentry key to what it will be before release. It's currently pointing to the ddev-loal-dev. All we have to do is change it in CircleCI
